### ISSUE TITLE
Extract demo out into separate repository

### ISSUE THREAD

**Issue by TimoGlastra** [hyperledger/aries-framework-javascript#1499]
The demo breaks quite often currently, as it needs to work with every commit into main. Extracting it means we only have to keep the demo working with stable releases from AFJ.

**Comment by J0rgeM** [comment]
To address frequent demo breaks, follow stable code practices, implement continuous integration, use separate development branches, versioning, and stable releases. Monitor and receive alerts for issues, ensuring a systematic approach to development, testing, and deployment while prioritizing code stability and quality.

**Comment by SurajAnand88** [comment]
After extracting , in which repository do we need to put the demo folder?

**Comment by blu3beri** [comment]
I am in favour of separating the demos to another repository. I think there is a major benefit in having quite some demos available and it would be bad to pollute the main repository with that. A small note on the side is that I think the demos should be version locked (maybe in a subfolder or something) and when we move to 0.5.x we copy the demo. This will keep a nice history and allow people on older versions to still see some examples. Since we don't push on older versions, maintenance would be non existent.

**Comment by Nas2020** [comment]
> I am in favour of separating the demos to another repository. I think there is a major benefit in having quite some demos available and it would be bad to pollute the main repository with that. A small note on the side is that I think the demos should be version locked (maybe in a subfolder or something) and when we move to 0.5.x we copy the demo. This will keep a nice history and allow people on older versions to still see some examples. Since we don't push on older versions, maintenance would be non existent.

@blu3beri 
To clarify the version locked ;  under the demos repo, there will be subfolder with the v0.4.0 release, and rest of the files. Would it be something like this?

````````````````````````````````````````````````````````````````````````
- aries-framework-javascript-demo /

  - v0.4.0 /
    - src /
      - Alice.ts
      - AliceInquirer.ts
      - BaseAgent.ts
      - BaseInquirer.ts
      - Faber.ts
      - FaberInquirer.ts
      - Listener.ts
      - OutputClass.ts
    - README.md (File)
    - package.json (File)
    - tsconfig.json (File)

````````````````````````````````````````````````````````````````````````

**Comment by blu3beri** [comment]
> > I am in favour of separating the demos to another repository. I think there is a major benefit in having quite some demos available and it would be bad to pollute the main repository with that. A small note on the side is that I think the demos should be version locked (maybe in a subfolder or something) and when we move to 0.5.x we copy the demo. This will keep a nice history and allow people on older versions to still see some examples. Since we don't push on older versions, maintenance would be non existent.
> 
> @blu3beri To clarify the version locked ; under the demos repo, there will be subfolder with the v0.4.0 release, and rest of the files. Would it be something like this?
> 
> ```
> - aries-framework-javascript-demo /
> 
>   - v0.4.0 /
>     - src /
>       - Alice.ts
>       - AliceInquirer.ts
>       - BaseAgent.ts
>       - BaseInquirer.ts
>       - Faber.ts
>       - FaberInquirer.ts
>       - Listener.ts
>       - OutputClass.ts
>     - README.md (File)
>     - package.json (File)
>     - tsconfig.json (File)
> ```

Yes, I think there should be a folder under `v0.4.0` with the name of the demo e.g. `AnoncredsWithDIDCommv1` or a better name.
